File: 2023/day01/part1.py
total = 0
for line in open("input.txt", "r"):
    for value in line:
        if value.isdigit():
            total += int(value)*10
            break
    for value in line[::-1]:
        if value.isdigit():
            total += int(value)
            break
        
print('Part 1:', total)

File: 2024/day01/part1.go
package main

import (
	"bufio"
	"bytes"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	var left []int
	var right []int

	input, err := os.ReadFile("input.txt") // Read input file
	if err != nil {
		fmt.Println("Error reading file:", err)
		return
	}

	scanner := bufio.NewScanner(bytes.NewReader(input)) // Create a scanner to read from the input
	for scanner.Scan() {
		line := scanner.Text()
		parts := strings.Split(line, "   ")
		fmt.Println(parts[0])
		}
	}